pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -273,10 +254,32 @@ func (g *Gatherer) createGatheringClosures(
 	return resultingClosures, errs
 }
 
+func (g *Gatherer) getGatheringRulesJSON(ctx context.Context) ([]byte, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, g.gatheringRulesEndpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := g.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	return ioutil.ReadAll(resp.Body)
+}
+
 // getConditionalGatheringFunctionName creates a name of the conditional gathering function adding the parameters
 // after the name. For example:
 //   "conditional/logs_of_namespace/namespace=openshift-cluster-samples-operator,tail_lines=100"
-func getConditionalGatheringFunctionName(funcName string, gatherParams map[string]interface{}) string {
+func getConditionalGatheringFunctionName(funcName string, gatherParamsInterface interface{}) string {
+	gatherParams, err := utils.StructToMap(gatherParamsInterface)
+	if err != nil {
+		// will only happen when non a struct is passed which means code is completely broken and panicking is ok
+		panic(err)
```

Comment:
I am affraid this can terminate our main go routine (that's undesirable....but correct me if I am wrong) so I think it would be safer to return the error.

Reply:
lemme check

Reply:
changed to returning an error